src/agent/rustjail/src/cgroups/systemd/subsystem/cpu.rs

```diff
@@ -0,0 +1,110 @@
+use super::super::common::{CgroupHierarchy, Properties};
+use super::transformer::Transformer;
+
+use anyhow::Result;
+use oci::{LinuxCpu, LinuxResources};
+use zbus::zvariant::Value;
+
+const BASIC_SYSTEMD_VERSION: &str = "242";
+const DEFAULT_CPUQUOTAPERIOD: u64 = 100 * 1000;
+const SEC2MICROSEC: u64 = 1000 * 1000;
+const BASIC_INTERVAL: u64 = 10 * 1000;
+
+pub struct Cpu {}
+
+impl Transformer for Cpu {
+    fn apply(
+        r: &LinuxResources,
+        properties: &mut Properties,
+        cgroup_hierarchy: &CgroupHierarchy,
+        systemd_version: &str,
+    ) -> Result<()> {
+        if let Some(cpu_resources) = &r.cpu {
+            match cgroup_hierarchy {
+                CgroupHierarchy::Legacy => Self::legacy_apply(cpu_resources, properties)?,
+                CgroupHierarchy::Unified => {
+                    Self::unified_apply(cpu_resources, properties, systemd_version)?
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Cpu {
+    // v1:
+    // cpu.shares <-> CPUShares
+    fn legacy_apply(cpu_resources: &LinuxCpu, properties: &mut Properties) -> Result<()> {
+        if let Some(shares) = cpu_resources.shares {
+            properties.push(("CPUShares", Value::U64(shares)));
+        }
+
+        Ok(())
+    }
+
+    // v2:
+    // cpu.shares <-> CPUShares
+    // cpu.period <-> CPUQuotaPeriodUSec
+    // cpu.period & cpu.quota <-> CPUQuotaPerSecUSec
+    fn unified_apply(
+        cpu_resources: &LinuxCpu,
+        properties: &mut Properties,
+        systemd_version: &str,
+    ) -> Result<()> {
+        if let Some(shares) = cpu_resources.shares {
+            let unified_shares = get_unified_cpushares(shares);
+            properties.push(("CPUShares", Value::U64(unified_shares)));
+        }
+        let quota = cpu_resources.quota.unwrap_or(i64::MAX);
```

Comment:
For an invalid value, it's better to return an `Err` instead of using a Max value or 0, IMO.